pennylane/_queuing_context.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the :class:`QueuingContext` abstract base class.
+"""
+import abc
+
+
+class QueuingContext(abc.ABC):
+    """Abstract base class for classes that exposes a queue for Operations.
+    """
+
+    _active_contexts = []
+    """The list of contexts that are currently active."""
+
+    def __enter__(self):
+        """Adds this instance to the global list of active contexts.
+
+        Returns:
+            QueuingContext: This instance
+        """
+        QueuingContext._active_contexts.append(self)
+
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        """Remove this instance from the global list of active contexts.
+        """
+        QueuingContext._active_contexts.remove(self)
+
+    @abc.abstractmethod
+    def _append_operator(self, operator):
+        """Append an operator to this QueuingContext instance.
+
+        Args:
+            operator (Operator): The Operator instance to be appended
+        """
+
+    @classmethod
+    def append_operator(cls, operator):
+        """Append an operator to the global queue(s).
+
+        Args:
+            operator (Operator): The Operator instance to be appended
+        """
+        for context in cls._active_contexts:
+            context._append_operator(operator)  # pylint: disable=protected-access
```

Comment:
Just clarifying, but this adds the operation to _all_ contexts in the context list?

Reply:
What is the use case you are envisioning here?

Reply:
Exactly! Active contexts are usually 0-1 qnodes and 0-X operation recorders, and I want the operation go to all of them.

Reply:
Perhaps this could be included in the docstring to make it clearer :)